streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -165,21 +167,27 @@ public void onCompletion(final RecordMetadata metadata,
                             if (exception instanceof ProducerFencedException) {
                                 log.warn(LOG_MESSAGE, key, value, timestamp, topic, exception.toString());
                                 sendException = new ProducerFencedException(
-                                    String.format(EXCEPTION_MESSAGE,
-                                                  logPrefix,
-                                                  "producer got fenced",
-                                                  key,
-                                                  value,
-                                                  timestamp,
-                                                  topic,
-                                                  exception.toString()));
+                                    String.format(
+                                        EXCEPTION_MESSAGE,
+                                        logPrefix,
+                                        "producer got fenced",
+                                        key,
+                                        value,
+                                        timestamp,
+                                        topic,
+                                        exception.toString()));
                             } else {
                                 if (productionExceptionIsFatal(exception)) {
                                     recordSendError(key, value, timestamp, topic, exception);
                                 } else if (productionExceptionHandler.handle(serializedRecord, exception) == ProductionExceptionHandlerResponse.FAIL) {
                                     recordSendError(key, value, timestamp, topic, exception);
                                 } else {
-                                    log.debug(HANDLER_CONTINUED_MESSAGE, key, value, timestamp, topic, exception);
+                                    log.warn(
+                                        "Error sending records (key=[{}] value=[{}] timestamp=[{}]) to topic=[{}] and partition=[{}]; " +
```

Comment:
Nit: do we need to `[]` around each value? `[]` is used for collections or list -- might be confusing to add them?

Reply:
Good question. I have developed the habit of delimiting variables in log messages, as it disambiguates the structure of the message for the reader. Without delimiters, there are several edge cases that would make the log message difficult to read.

For example, if the key were `"value"` and the value were `""` with the old format, you get:
```
Error sending records (key value value  timestamp 1234)
```
Whereas, if the key were `""` and the value were `"value"`, you get
```
Error sending records (key  value value timestamp 1234)
```
The only difference between these strings is where the extra space is. With delimiters, you have:
```
Error sending records (key=[value] value=[] timestamp=[1234])
Error sending records (key=[] value=[value] timestamp=[1234])
```

It's the kind of thing that saves people from #1 making a bad assumption about the nature of the problem and burning hours before they realize their mistake, or #2 being unable to clearly understand the error message and having to load it in a debugger just to understand what the values of the arguments actually are.

It sounds like your concern is about the ambiguity of `[]` as delimiters, since they already indicate a list. Can we keep delimiters but pick a different character? Other paired delimiters are `<>` and `{}`, and `""` and `''` also come to mind. WDYT?

Reply:
I personally thank, that having `=` (without `[]`) is good enough as the `=` makes it clear:
```
Error sending records (key=value value= timestamp=1234)
```
Thus, it's not ambiguous to me (I agree that having no delimiter at all would be bad).

It's just that I like uniform formatting, and this would introduce a new style -- I am fine with change to this style, but we should agree on one style and rewrite code (on the side) if it does not fit the 'style guide'.

\cc @bbejeck @dguy @guozhangwang

Reply:
I tend to prefer `key=[value]`, but I do not have a scientific reason for that: I just feel it is more "vivid" :P

Reply:
@mjsax I can dig the desire to have uniform style on log messages. I'll also point out that the logs are part of the public API, so we can't just go terraform them willy-nilly, but instead we'd have to change them only in scope of the relevant KIPs, which makes it difficult to change, or even establish, a log style.

Nevertheless, if we don't already have a clear style for streams logs, I'll advocate for some kind of enclosing delimiter on substitutions. I continue to agree that square brackets are confusing w.r.t. common `List#toString()` formats, so I think we should agree on a different enclosing delimiter.

I agree that `=` is better than nothing, but it's still ambiguous when the substitution is 0 or more whitespace characters, while `[]` vs `[ ]` gives you more of a clue. No choice here is going to be perfect, but my experience is that this format saves enough debugging time to be worth the visual noise.

Reply:
> if we don't already have a clear style for streams logs

We never discussed this explicitly; it's just a matter of fact that we use `key=value` so far from what I can remember. The question is, how much we gain if we start to rewrite to a different format and how much work it it.

With regard to ambiguity: you can always construct an (academic?) example for which any formatting strategy "break" and is ambiguous... If we agree on `key=[value]` I am fine with it. Still not sure, if we gain much (but if you think we do, it's fine with me to change)

Reply:
I also prefer `key=[value]`, but can't say it's for any specific reason other than personal preference.